vars/vars_tracker.go

```diff
@@ -13,35 +14,64 @@ type CredVarsTrackerIterator interface {
 	YieldCred(string, string)
 }
 
+//go:generate counterfeiter . CredVarsTracker
+
 type CredVarsTracker interface {
 	Variables
 	IterateInterpolatedCreds(iter CredVarsTrackerIterator)
 	Enabled() bool
+
+	AddLocalVar(string, interface{}, bool)
 }
 
 func NewCredVarsTracker(credVars Variables, on bool) CredVarsTracker {
-	if on {
-		return &credVarsTracker{
-			credVars:          credVars,
-			interpolatedCreds: map[string]string{},
-			lock:              sync.RWMutex{},
-		}
-	} else {
-		return dummyCredVarsTracker{credVars: credVars}
+	return &credVarsTracker{
+		localVars:         StaticVariables{},
+		credVars:          credVars,
+		enabled:           on,
+		interpolatedCreds: map[string]string{},
+		noRedactVarNames:  map[string]bool{},
+		lock:              sync.RWMutex{},
 	}
 }
 
 type credVarsTracker struct {
-	credVars          Variables
+	credVars  Variables
+	localVars StaticVariables
+
+	enabled bool
+
 	interpolatedCreds map[string]string
 
+	noRedactVarNames map[string]bool
+
 	// Considering in-parallel steps, a lock is need.
 	lock sync.RWMutex
 }
 
 func (t *credVarsTracker) Get(varDef VariableDefinition) (interface{}, bool, error) {
-	val, found, err := t.credVars.Get(varDef)
-	if found {
+	var (
+		val   interface{}
+		found bool
+		err   error
+	)
+
+	redact := true
+	parts := strings.Split(varDef.Name, ":")
+	if len(parts) == 2 && parts[0] == "." {
```

Comment:
Maybe `credVars` could have an entry within its `NamedVariables` whose name is `.`, and have the same entry passed to the `*credVarsTracker` so it can directly set vars in it? I think that way we can avoid both this change and the change to `vars/template.go` that I commented on.